release-notes/Cabal-3.16.0.0.md

```diff
@@ -0,0 +1,202 @@
+Cabal and Cabal-syntax 3.16.0.0 changelog and release notes
+---
+
+### Significant changes
+
+- Make `Flag a` a type synonym for `Last (Maybe a)` [#10948](https://github.com/haskell/cabal/pull/10948)
+
+    Refer to the [Migration guide](#migration-guide) for an upgrade path.
+
+- Add support for Windows Aarch64 [#10705](https://github.com/haskell/cabal/pull/10705)
+
+  Adds to preprocessor branches the option to support the `aarch64_HOST_ARCH` platform on the Windows Aarch64 target.
+  The `ccall` convention is used on Aarch64, as with `x86_64_HOST_ARCH`. Introduce `zIsAarch64` to make paths generation support the Windows Aarch64 target.
+
+### Other changes
+
+- Add --with-repl flag to specify alternative REPL program [#9115](https://github.com/haskell/cabal/issues/9115) [#10996](https://github.com/haskell/cabal/pull/10996)
+
+  Added a new `--with-repl` command-line option that allows specifying an alternative
+  program to use when starting a REPL session, instead of the default GHC.
+
+  This is particularly useful for tools like `doctest` and `hie-bios` that need to
+  intercept the REPL session to perform their own operations. Previously, these tools
+  had to use `--with-ghc` which required them to proxy all GHC invocations, including
+  dependency compilation, making the implementation more complex.
+
+  The `--with-repl` option only affects the final REPL invocation, simplifying the
+  implementation of such wrapper tools.
+
+  Example usage:
+  ```bash
+  cabal repl --with-repl=doctest
+  cabal repl --with-repl=/path/to/custom/ghc
+  ```
+
+  This change also removes the special handling for response files with `--interactive`
+  mode, as tools are now expected to handle response files appropriately.
+
+- Don't imply that Haddock docs exist, when warning not installed [#9694](https://github.com/haskell/cabal/issues/9694) [#9695](https://github.com/haskell/cabal/pull/9695)
+
+- Show why `cabal act-as-setup configure` failed [#10273](https://github.com/haskell/cabal/pull/10273)
+
+  When `cabal act-as-setup configure` fails, it prints a list of "missing or
+  private dependencies".
```

Comment:
Today I learned that `cabal act-as-setup` is a thing. It looks a little strange to mention `cabal` (the tool) in the changelog for Cabal-the-library; at least in th way it's done here --- implying that `cabal` is the main target of the change. Since we're in the context of Cabal-the-lib by default in these notes, I suggest:

```suggestion
- Show why the configure step failed [#10273](https://github.com/haskell/cabal/pull/10273)

  When configuring fails, we print a list of "missing or private dependencies".
```